pkg/metrics/metrics.jsx

```diff
@@ -391,37 +399,62 @@ class CurrentMetrics extends React.Component {
                     const is_user = k.match(/^user.*user@\d+\.service.+/);
                     const label = k.replace(/.*\//, '').replace(/\.service$/, '');
                     // only keep cgroup basenames, and drop redundant .service suffix
-                    merged.push([label, v, is_user, false]);
+                    merged.push([label, v, is_user, false, k]);
                 }
                 // filter out podman containers
                 const matches = k.match(podmanCgroupRe);
                 if (matches && v) {
-                    // truncate to 12 chars like the podman output
-                    const containerid = matches.groups.containerid.substr(0, 12);
-                    const is_user = k.match(/^user.slice/);
-                    merged.push([containerid, v, is_user, true]);
+                    let is_user = false;
+                    let uid = '0';
+                    const containerid = matches.groups.containerid;
+                    const umatches = k.match(useridCgroupRe);
+                    if (umatches) {
+                        is_user = true;
+                        uid = umatches.groups.userid;
+                    }
+                    merged.push([containerid, v, is_user, true, uid]);
                 }
             });
             merged.sort((a, b) => b[1] - a[1]);
             return merged.slice(0, n);
         }
 
-        function cgroupClickHandler(name, is_user, is_container) {
+        function podNameInCache(podNameMapping, uid, containerid) {
+            if (uid in podNameMapping) {
+                return containerid in podNameMapping[uid];
+            } else {
+                return false;
+            }
+        }
+
+        function podName(podNameMapping, uid, containerid) {
+            if (podNameInCache(podNameMapping, uid, containerid)) {
+                return podNameMapping[uid][containerid];
+            } else {
+                return containerid.substr(0, 12);
+            }
+        }
+
+        function cgroupClickHandler(name, is_user, is_container, uid, podNameMapping) {
             if (is_container) {
-                cockpit.jump("/podman");
+                if (podNameInCache(podNameMapping, uid, name)) {
+                    cockpit.jump("/podman#/?name=" + podName(podNameMapping, uid, name));
```

Comment:
This would show a link for users which can not be viewed in cockpit-podman for example:

I have a user named `jelle` who runs a container, with administrative access I can see their cgroups but cockpit-podman only shows your containers + administrative access containers, so the links don't really work.

As we know the userid, we could check if the uid is 0 or current logged and then show a link. Otherwise show for who the container owner is (that would require a username lookup) and ofcourse has to be done in cgroupRow.

Reply:
I'd actually prefer to not show these at all. If you still want to, I don't veto this, though -- then for that I like your suggestion.

Reply:
I'm ok with leaving it out, having a follow up which explains what `user@1000` is, so ideally it would be: `user session for jelle` but maybe that's too long already.

Reply:
How about "user jelle", similar to "pod foo"? But yes, this is follow-up material, as it's unrelated to this PR.